modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -439,6 +439,36 @@ def take_2d_labels_or_positional(
             return base
 
         row_positions = maybe_range(row_positions)
+
+        # If row_positions is not range, then MaskNode will generate a filter,
+        # containing enumeration of all the positions. Filtering rows in this
+        # way is not efficient and, in casse of too many values in row_positions,
+        # may result in a huge JSON query. To workaround this issue, creating an
+        # empty frame with row_positions index and inner joining with this one.
+        # If row_positions has less than 10 values, MaskNode is used.
+        if (
```

Comment:
Should we consider this PR as a PERF enhancement or leave as a FIX? If 2nd, should we add a test for the issue?

Reply:
test_iloc_int() covers this case.